lib/order.rb

```diff
@@ -0,0 +1,76 @@
+require 'csv'
+require_relative 'customer'
+
+class Order
+  attr_reader :id
+  attr_accessor :products, :customer, :fulfillment_status
+
+  VALID_STATUS = [:pending, :paid, :processing, :shipped, :complete]
+
+  def initialize(id, products, customer, fulfillment_status = :pending)
+    @id = id
+    @products = products
+    @customer = customer
+    if VALID_STATUS.include?(fulfillment_status)
+      @fulfillment_status = fulfillment_status
+    else
+      raise ArgumentError, "#{fulfillment_status} is not a valid status."
+    end
+  end
+
+  def total
+    tax = 0.075
+    if products.empty?
+      return 0
+    else
+      total_bill = (products.values.reduce(:+) * (1 + tax)).round(2)
+      return total_bill
+    end
+  end
```

Comment:
This is a very small suggestion, but I thought it's fun...

You did great logic for saying "if we have no products, we should return `0`!" That's good logic!

However... when `products` is an empty hash... if did the rest of the method anyway (starting on line 26)... when we sum up all of the values, it'll turn out to be `0`, it is still accurate to our logic. Therefore, there's a way to write this method so you don't check if products is empty.

You'd have to change your logic and not use `reduce`, but consider the following code as a possible refactoring:
```suggestion
  def total
    tax = 0.075
    total_bill = (products.values.sum * (1 + tax)).round(2)
    return total_bill
  end
```

Reply:
Oh wow.. this is also good to know!!  I love these reviews, thank you!